echopype/consolidate/api.py

```diff
@@ -364,9 +369,19 @@ def add_splitbeam_angle(
             theta, phi = get_angle_complex_samples(ds_beam, angle_params)
 
     # add theta and phi to source_Sv input
-    source_Sv = add_angle_to_ds(
-        theta, phi, source_Sv, return_dataset, source_Sv_path, file_type, storage_options
-    )
+    theta.attrs["long_name"] = "split-beam alongship angle"
+    phi.attrs["long_name"] = "split-beam athwartship angle"
+
+    # add the split-beam angles to the provided Dataset
+    source_Sv["angle_alongship"] = theta
+    source_Sv["angle_athwartship"] = phi
+    if to_disk:
+        if source_Sv_type is not None:
```

Comment:
I think there should be a check up front, right after `source_Sv_type = get_file_format(source_Sv)`, where you fail the case where `to_disk=True and source_Sv_type is None`, otherwise it is not obvious what's going on if people put `to_disk=True` but the to_netcdf/to_zarr are not invoked due to `source_Sv_type` being None.

Reply:
Actually, looking at `get_file_format`, it does not seem like the output would ever be None. Could you rework the logic up front? What I wanted to say in this and the previous comment is that, if people put in a `source_Sv` that is an in-memory object instead of a path, an error should be raised if they also set `to_disk=True`, because there is no place the data can be saved to.

Reply:
@leewujung Understood, an error will be raised if `source_Sv` is not a path and `to_disk` is true.